src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -169,6 +171,8 @@ public void Initialize(bool dbPerWallet = true)
                     this.Wallets[wallet.Name] = walletContainer;
 
                     this.logger.LogDebug("Added '{0}` to wallet collection.", wallet.Name);
+
+                    this.RemoveAllUnconfirmedTransactions(wallet.Name);
```

Comment:
Need to know how this fix works. Maybe a comment at the least.

The node starts up. There is a mempool.dat. Those transactions are loaded into the mempool - Does that trigger `ProcessTransaction` again?

Reply:
The idea is to simply remove all unconfirmed transactions on startup. The transactions can now arrive via the blocks, so any updates will now be delayed until that happens.

That may be sufficient - as it was for cross chain transfers. However, if not, I can re-load the unconfirmed transactions into the wallet based on the transactions present in the mempool.